react/components/ButtonPlain/index.js

```diff
@@ -0,0 +1,248 @@
+import React, { Component, Fragment } from 'react'
+import PropTypes from 'prop-types'
+
+import Spinner from '../Spinner'
+
+import { withForwardedRef, refShape } from '../../modules/withForwardedRef'
+
+class ButtonPlain extends Component {
+  handleClick = event => {
+    !this.props.disabled &&
+      !this.props.isLoading &&
+      this.props.onClick &&
+      this.props.onClick(event)
+  }
+
+  componentDidMount() {
+    if (this.props.icon) {
+      console.warn(
+        'Button: The prop "icon" of the "Button" component has been deprecated, and will be removed in a future version. Please use the component "ButtonWithIcon" instead'
+      )
+    }
+  }
+
+  render() {
+    const {
+      size,
+      variation,
+      children,
+      icon,
+      isLoading,
+      href,
+      target,
+      rel,
+      referrerPolicy,
+      download,
+      noWrap,
+    } = this.props
+
+    const disabled = this.props.disabled || isLoading
+    const iconOnly = icon || this.props.iconOnly
+    const horizontalPadding = iconOnly ? 1 : 2
+    const horizontalCompensation = `nr${horizontalPadding} nl${horizontalPadding} `
+
+    let classes = 'vtex-button bw1 ba fw5 v-mid relative pa0 br2 bn '
```

Comment:
I'm not a fan of attaching classes latter on for two main reasons:
1. The code becomes spaghetti. Meaning that I need to scroll (up and down) to figure out how the `classes` final value would be.
2. If someone forgot any space at strings tail... Boom 💣! We got a bug in our hands.

Since there is a lot of constraints, using template strings would not be any better (will become a total mess). For that use case, I think `classnames` is very useful.